### PR TITLE
Refactor 'alpha' bool to an 'itemCategory' enum value.

### PR DESCRIPTION
**Pull request by maryla-uc**
This refactor will be useful when adding gain maps.

### CODE REVIEW

File: src/read.c [hunk 1]
```diff
@@ -4194,6 +4194,11 @@ static avifResult avifImageLimitedToFullAlpha(avifImage * image)
     return AVIF_RESULT_OK;
 }
 
+static avifResult avifGetErrorForItemType(avifItemType itemType)
```
Comment (y-guyon) [inline src/read.c:4197]: Declare it in internal.h next to `avifItemType` and define it only once?

File: include/avif/internal.h [hunk 2]
```diff
@@ -262,8 +271,8 @@ AVIF_ARRAY_DECLARE(avifDecodeSampleArray, avifDecodeSample, sample);
 typedef struct avifCodecDecodeInput
```
[the remark this answers is not in the document]
Comment (maryla-uc) [reply include/avif/internal.h:271]: Indeed, but I think it should be done in another PR.

File: src/write.c [hunk 3]
```diff
@@ -157,15 +157,15 @@ void avifCodecEncodeOutputDestroy(avifCodecEncodeOutput * encodeOutput)
...
+    uint8_t type[4];                      // 4-character ISOBMFF box type
     avifCodec * codec;                    // only present on image items
     avifCodecEncodeOutput * encodeOutput; // AV1 sample data
     avifRWData metadataPayload;           // Exif/XMP data
     avifCodecConfigurationBox av1C;       // Harvested in avifEncoderFinish(), if encodeOutput has samples
                                           // TODO(yguyon): Rename or add av2C
     uint32_t cellIndex;                   // Which row-major cell index corresponds to this item. only present on image items
-    avifBool alpha;
-    avifBool hiddenImage; // A hidden image item has (flags & 1) equal to 1 in its ItemInfoEntry.
+    avifItemType itemType;                // Type of item being encoded
```
[the remark this answers is not in the document]
Comment (maryla-uc) [reply src/write.c]: Changed to 'itemCategory'

File: src/write.c [hunk 4]
```diff
@@ -902,17 +902,17 @@ static avifResult avifEncoderAddImageItems(avifEncoder * encoder,
                                            uint32_t gridRows,
                                            uint32_t gridWidth,
                                            uint32_t gridHeight,
-                                           avifBool alpha,
+                                           avifItemType itemType,
                                            uint16_t * topLevelItemID)
 {
     const uint32_t cellCount = gridCols * gridRows;
-    const char * infeName = alpha ? "Alpha" : "Color";
+    const char * infeName = (itemType == AVIF_ITEM_ALPHA) ? "Alpha" : "Color";
```
[the remark this answers is not in the document]
Comment (maryla-uc) [reply src/write.c]: It's only used in one place so I think it's fine as is?

**Review by y-guyon — APPROVED** [review]

**Pull request merged**